### ISSUE TITLE
build failure

### ISSUE THREAD

**Issue by thesrinivas** [tumblr/collins#439]
I see this build failure following the steps (using JDK1.7). Anyone knows what I'm missing?

```
    ==== Maven Central: tried

      http://repo1.maven.org/maven2/org/scala-sbt/sbt_2.9.1/0.13.5/sbt_2.9.1-0.13.5.pom

            ::::::::::::::::::::::::::::::::::::::::::::::

            ::          UNRESOLVED DEPENDENCIES         ::

            ::::::::::::::::::::::::::::::::::::::::::::::

            :: org.scala-sbt#sbt_2.9.1;0.13.5: not found

            ::::::::::::::::::::::::::::::::::::::::::::::
```

:: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
unresolved dependency: org.scala-sbt#sbt_2.9.1;0.13.5: not found
Error during sbt execution: Error retrieving required libraries

**Comment by Primer42** [comment]
@thesrinivas can you specify which version of collins you are trying to build and which build instructions you are following? There are a few ways to build, and the docs are in a mixed state betweenthe last release and the next release, as we are trying update the docs before we publish the next release.

**Comment by thesrinivas** [comment]
I followed steps as listed in https://tumblr.github.io/collins/#quickstart-source under "...build it yourself". The git source is pointing to the latest, I presume.

I'm new to collins. Would you have some more docs for various use-cases supported? In particular, I'm interested in seeing changes for a particular ASSET.

Thanks

**Comment by Primer42** [comment]
@thesrinivas - it failed because the latest master is on play 2.3.
Give this a try instead:
- download the minimal package for play activator 1.3.6 from [here](https://downloads.typesafe.com/typesafe-activator/1.3.6/typesafe-activator-1.3.6-minimal.zip). Unpackage it into `~/src/activator-1.3.6-minimal`
- execute `~/src/activator-1.3.6-minimal/activator/run`
I just did that on the latest master, and was able to get it to work using java 8 - I'm trying now with java 7, but it takes a while to build.

Can you explain what you mean by "seeing changes for a particular asset"?

In short, collins primarily is used to store information about physical assets in your infrastructure, with a robust API that allows you to build complicated automation to manage your servers.  Once you have collins populated with information about your servers, the possibilities are fairly endless. At tumblr, we use it to drive our configuration management, config file generation for things like DNS and determine what monitoring each server needs - pretty much every project our SRE team does involves collins in some way.

**Comment by Primer42** [comment]
@thesrinivas - I was just able to build using java 7 and the instructions I laid out - please let me know if they work for you.

**Comment by thesrinivas** [comment]
I had java8 but installed java7 as per the docs. I'll revert to go back to
java 8.

Change management and actions based on the change is an important use-case
here.

Here's a simple example that I'd like to show-case (for others here). Since
I'm new, I'd appreciate if you could give an example of how/if I could
accomplish this.

Say there are 10 assets, I'm interested in attribute fields (ex. "Amount of
Memory") that are different.

Thanks

**Comment by Primer42** [comment]
@thesrinivas - with the newer version of play, java 8 works.

Collins doesn't have a view like that built in - it's really meant to be as simple as possible, and allow you to do what you want.  That said, the information is all there - it would not be hard to query the API and create that sort of diff.

**Comment by thesrinivas** [comment]
hmmm. There is no activator/run in that zip.

**Comment by thesrinivas** [comment]
I dont see ~/src/activator-1.3.6-minimal/activator/run script in the zip
file.
Can you please check?

**Comment by thesrinivas** [comment]
@Primer42 
Using lshw output from a Suse12 system, I get this error. What am I missing?

       " message Error updating asset: Could not find capacity for network interface for MT27520 Family [ConnectX-3 Pro] [15B3:1007]"

**Comment by byxorna** [comment]
@thesrinivas you will need to do some xml postprocessing on your LSHW output. Collins expects any interfaces listed to have `<capacity>12345</capacity>`. If lshw is too dumb to discover that, you will have to inject it yourself before submitting the hardware report. (Same with vlan names in LLDP xml).

**Comment by thesrinivas** [comment]
Can you list a couple of lshw and lldp commands with the arguments that are
typically used to enter into collins.
I see that the lldp package doesnt always do xml output. Do you have the
steps to install lldp?
Having these steps documented helps avoid lot of wasted time for newbies
like myself.

Also, can you respond to the question about the missing activator/run
command that I asked earlier in this thread?

Thanks

**Comment by Primer42** [comment]
We just run `lshw -quiet -xml` and `lldpctl", "-f", "xml`.

You don't execute `activator/run` you execute `activator run` - it's a subcommand.  My mistake, I made a typo.  Sorry about that.

Hope that helps

**Comment by byxorna** [comment]
It is complicated by the fact that there is no `lldpd` rpm packaged up in epel or firstparty centos/rhel/SL repos. @Primer42 we should look at bundling lldpd (and lshw?) into the genesis image to make it more turnkey.

**Comment by thesrinivas** [comment]
Thanks. Similar to LSHW and LLDP, I'm working on adding support for a BIOS
config XML (reported by HP's conrep utility).
Once the data is entered, I need to be able to query/diff/track various
fields of interest.
Can you please send me any pointers on this?

**Comment by byxorna** [comment]
I am going to mark this as closed. Thanks!
